internal/commitremap/commitremap.go

```diff
@@ -1,129 +1,169 @@
 package commitremap
 
 import (
+	"bufio"
+	"bytes"
 	"encoding/json"
 	"fmt"
 	"log"
 	"os"
 	"path/filepath"
 	"strings"
+	"sync"
 )
 
-// Struct to represent a single entry in the commit map
-type CommitMapEntry struct {
-	Old string
-	New string
+type File struct {
+	FilePath string
+	Prefix   string
 }
 
-// Parses the file and returns a map of old commit hashes to new commit hashes
-func ParseCommitMap(filePath string) (*[]CommitMapEntry, error) {
-	commitMap := []CommitMapEntry{}
+// Parses the commit-map file and returns a map of old commit hashes to
+// new commit hashes using the old commit sha as the key
 
-	// Read the commit-map file
+func ParseCommitMap(filePath string) (*map[string]string, error) {
+	commitMap := make(map[string]string)
 	content, err := os.ReadFile(filePath)
 	if err != nil {
 		return nil, err
 	}
-	// Split the file content into lines
-	lines := strings.Split(string(content), "\n")
-
-	// Iterate over the lines and parse the old and new commit hashes
-	for _, line := range lines {
-		if strings.TrimSpace(line) == "" {
+	buf := bytes.NewBuffer(content)
+	scanner := bufio.NewScanner(buf)
+	for scanner.Scan() {
+		line := scanner.Text()
+		// Skip adding the header to the map
+		if line == "old                                      new" {
 			continue
 		}
-
-		fields := strings.Fields(line)
+		fields := strings.Split(line, " ")
 		if len(fields) != 2 {
 			return nil, fmt.Errorf("invalid line: %s", line)
 		}
-
-		commitMap = append(commitMap, CommitMapEntry{
-			Old: fields[0],
-			New: fields[1],
-		})
+		commitMap[fields[0]] = fields[1]
```

Comment:
To improve readability we should be explicit about what the fields are mapping to, something like:
        oldSha, newSha := fields[0], fields[1]
        commitMap[oldSha] = newSha

Reply:
👍